js/onto2bsdd.js

```diff
@@ -68,42 +90,43 @@ class Onto2bsdd {
               ? "Time"
               : "String";
           resultPropertyObject = {
-            Code: Onto2bsdd.getLocalname(csvObject.ontoPropertyURI),
+            Code: Onto2bsdd.getLocalname(ontoPropertyURI),
             DataType: bsddDatatype,
             Definition: csvObject.ontoPropertyDefinition,
             Name: csvObject.ontoPropertyPrefLabel,
-            OwnedUri: csvObject.ontoPropertyURI,
+            OwnedUri: ontoPropertyURI,
           };
           resultProperties.push(resultPropertyObject);
         }
 
-        const resultPropertyLinkObject = {
-          Code: md5(
-            resultClassificationObject.Code + "-" + resultPropertyObject.Code
-          ),
+        const classProperty = {
+          // Code: md5(
+          //   resultClassificationObject.Code + "-" + resultPropertyObject.Code
+          // ),
+          Code: resultPropertyObject.Code,
           PropertyCode: resultPropertyObject.Code,
           PropertySet: result.DictionaryCode,
           PropertyType: "Property",
         };
-        if (csvObject.ontoPropertyURI.includes(dictionaryUri)) {
-          resultPropertyLinkObject.OwnedUri = csvObject.ontoPropertyURI;
+        if (ontoPropertyURI.includes(dictionaryUri)) {
+          classProperty.OwnedUri = ontoPropertyURI;
         } else {
-          resultPropertyLinkObject.OwnedUri = dictionaryUri + "fake-uri-404";
+          classProperty.OwnedUri =
+            dictionaryUri + "class-property-fake-uri-404";
         }
-        resultClassificationObject.ClassProperties.push(
-          resultPropertyLinkObject
-        );
+        resultClassificationObject.ClassProperties.push(classProperty);
       }
 
       if (csvObject.mappedClassURI && csvObject.mappedClassRelation) {
-        const resultIfcRelationObject = {
+        const classRelation = {
           RelationType: csvObject.mappedClassRelation,
-          RelatedClassUri: csvObject.mappedClassURI,
+          RelatedClassUri: replaceIfcUri(csvObject.mappedClassURI),

```

Comment:
Rename function to replaceIfcNamespace. Or better yet, use a single function called replaceDeprecatedNamespaces that takes an array of namespace pairs (deprecated & new) that are to be replaced. I expect more such deprecations in the future, especially with people joining in who are unfamiliar with the use of (stable) URIs.

Reply:
moved namespace replacement back to sparql